server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -410,16 +409,24 @@ static ClusterState closeRoutingTable(final ClusterState currentState,
                 }
 
                 logger.debug("closing index {} succeeded", index);
-                blocks.removeIndexBlockWithId(index.getName(), INDEX_CLOSED_BLOCK_ID).addIndexBlock(index.getName(), INDEX_CLOSED_BLOCK);
-                metadata.put(IndexMetaData.builder(indexMetaData).state(IndexMetaData.State.CLOSE));
-                routingTable.remove(index.getName());
-                closedIndices.add(index.getName());
+                final IndexMetaData updatedIndexMetaData = IndexMetaData.builder(indexMetaData).state(IndexMetaData.State.CLOSE).build();
+                metadata.put(updatedIndexMetaData, true);
```

Comment:
why change this to use `boolean incrementVersion`? Isn't that already done with the regular put?

Reply:
Right, I messed up things.